scspell/__init__.py

```diff
@@ -691,8 +691,17 @@ def spell_check(source_filenames, override_dictionary=None,
     with CorporaFile(dict_file, base_dicts, relative_to) as dicts:
         ignores = set()
         for f in source_filenames:
-            if not spell_check_file(f, dicts, ignores, report_only, c_escapes):
-                okay = False
+            if not os.path.isdir(f):
+                if not spell_check_file(f, dicts, ignores, report_only, c_escapes):
+                    okay = False
+            else:
+                for dir, subdir, files in os.walk(f):
```

Comment:
Same comment as @jayvdb about `dir` being reserved. I see that the code changed, but the name collision is still there.